src/main/java/cms/gov/madie/measure/utils/JsonUtil.java

```diff
@@ -632,4 +644,53 @@ public static String getTestCaseJson(String json) throws JsonProcessingException
     }
     return null;
   }
+
+  // going through JsonNode, converts any datetime string into UTC datetime
+  public static JsonNode replaceNestedDateTimeStringValue(JsonNode node) {
+    if (node.isObject()) {
+      ObjectNode objectNode = (ObjectNode) node;
+      objectNode
+          .fields()
+          .forEachRemaining(
+              entry -> {
+                String fieldName = entry.getKey();
+                JsonNode childNode = entry.getValue();
+                String currentValue = childNode.asText();
+                String newValue = getNewValue(currentValue);
+                if (childNode.isTextual()) {
+                  objectNode.put(fieldName, newValue);
+                } else {
+                  replaceNestedDateTimeStringValue(childNode);
+                }
+              });
+    } else if (node.isArray()) {
+      node.forEach(childNode -> replaceNestedDateTimeStringValue(childNode));
+    }
+    return node;
+  }
+
+  // converts a value into UTC datetime, if the the passed in value is a datetime
+  // otherwise, returns the original value
+  static String getNewValue(String value) {
+    String newValue = value;
+    if (value.length() > 19 && (PATTERN.matcher(value.substring(0, 19)).matches())) {
+      ZonedDateTime dateTime = null;
+      ZonedDateTime adjustedDateTime = null;
+      try {
+        dateTime = ZonedDateTime.parse(value, FORMATTER);
+        adjustedDateTime = dateTime.withZoneSameInstant(ZoneId.of("UTC"));
+        newValue = adjustedDateTime.format(FORMATTER).replace("Z", "+00:00");
+      } catch (DateTimeParseException e) {
+        try {
+          // for older data without milliseconds
+          dateTime = ZonedDateTime.parse(value, FORMATTER_2);
+          adjustedDateTime = dateTime.withZoneSameInstant(ZoneId.of("UTC"));
+          newValue = adjustedDateTime.format(FORMATTER_2).replace("Z", ".000+00:00");
+        } catch (DateTimeParseException ex) {
+          log.info("Error parsing date/time string: " + e.getMessage());
```

Comment:
Should this be an error ? and should we allow user to continue if an exception is encountered.

Reply:
@RohitKandimalla Yeah it should be an error. I think we should allow user to continue if there is something wrong with a datetime, what do you think?

Reply:
@RohitKandimalla Nicole said we should allow it.

Reply:
Then I believe this should atleast be at WARN level

Reply:
@RohitKandimalla Ok changed to warn.